src/statistic/statistic.js

```diff
@@ -0,0 +1,131 @@
+     * @returns {Promise<*[]>} an Array containing all the stats of the admin
+     */
+    async adminStats(aUsername){
+        let allDiscussions = await dbManager.getDiscussionsAdmin(aUsername);
+
+        return [allDiscussions.length, Array.from(allDiscussions.values(), d => this.discussionFormat(d))];
+    };
+
+    /**
+     * Get the number of the total of votes for all the responses for a question.
+     * @param aQuestionId integer that is the id of the question
+     * @param aDiscussionId integer that is the id of the discussion
+     * @returns {Promise<number>} an integer that is the number of votes
+     */
+    async getNumberVotesQuestion(aQuestionId, aDiscussionId) {
+        // Get all the responses related to the question desired
+        let allResponses = await dbManager.getResponsesQuestion(aQuestionId, aDiscussionId);
+        let numberTotalVotes = 0;
+        // Get all the votes for all the responses
+        for (let i = 0; i < allResponses.length; ++i) {
+            numberTotalVotes += allResponses[i].devices.length;
+        }
+        logger.info(`The number of votes : ${numberTotalVotes}`);
+        return numberTotalVotes;
+    }
+
+    /**
+     * Get all the stats for a debate
+     * @param aDiscussionId the id of the discussion to get the stats from
+     * @returns {Promise<*[]>} an array containing the stats wanted for a debate
+     */
+    async debateStats(aDiscussionId){
+        // Get the discussion in the database
+        let debate = await dbManager.getDiscussion(aDiscussionId);
+        // Get all the questions related to the debate
+        let allQuestions = await dbManager.getQuestionsDiscussion(aDiscussionId);
+        // Get the number of votes for all the Questions
+        for(let i = 0; i < allQuestions.length; ++i) {
+            allQuestions[i].numberVotes = await this.getNumberVotesQuestion(allQuestions[i].id, aDiscussionId);
+        }
+
+        return [allQuestions.length, debate.auditors, Array.from(allQuestions.values(), q => this.questionFormat(q)).
+        // Will sort the questions by the number of votes most actives in first places
+        sort(function (a, b) {
+            if (a.numberVotes > b.numberVotes) {
+                return -1;
+            }
+            if (b.numberVotes > a.numberVotes) {
+                return 1;
+            }
+            return 0;
+        })];
+    };
+
+    /**
+     * Get all the stats a the question
+     * @param aQuestionId integer that is the id of the question
+     * @param aDiscussionId integer that is the id of the discussion
+     * @returns {Promise<*[]>} Array that contains the stats wanted for a question
+     */
+   async questionStats(aQuestionId, aDiscussionId){
+       let debate = await dbManager.getDiscussion(aDiscussionId);
+       let allResponses = await dbManager.getResponsesQuestion(aQuestionId, aDiscussionId);
+       let numberTotalVotes = await this.getNumberVotesQuestion(aQuestionId,aDiscussionId);
```

Comment:
Same here. Can `dbManager.getDiscussion`, `dbManager.getResponsesQuestion` or `this.getNumberVotesQuestion` return a null object ?

Reply:
As said before corrected in the new commit